File: app/src/main/java/com/ryanmounla/mvvmarch/data/EndpointService.java
package com.ryanmounla.mvvmarch.data;

import retrofit2.Call;
import retrofit2.http.GET;

/**
 * Created by ryanmounla on 2018-05-28.
 */

public interface EndpointService {

    @GET("/api/breeds/image/random")
    Call<DogImageResponse> getRandomDogImage();
}

File: app/src/main/java/com/ryanmounla/mvvmarch/data/EndpointService.kt
package com.ryanmounla.mvvmarch.data

import retrofit2.Call
import retrofit2.http.GET

/**
 * Created by ryanmounla on 2018-05-28.
 */

interface EndpointService {

    @get:GET("/api/breeds/image/random")
    val randomDogImage: Call<DogImageResponse>
}
